rfcs/framework-0072-pluggable-transfers.md

```diff
@@ -0,0 +1,507 @@
+    uploaderComponent: UploaderComponent,  // React component for uploader field
+  };
+```
+
+A complete Files UI entrypoint would look like:
+```js
+// ./js/invenio_app_rdm/deposit/files-uppy.js
+import { getInputFromDOM } from '@js/invenio_rdm_records';
+import { UppyDepositFileApiClient, UppyDepositFilesService, UppyUploader } from '@inveniosoftware/invenio-files-uppy';
+
+const {apiHeaders, default_transfer_type: defaultTransferType, fileUploadConcurrency} = getInputFromDOM("deposits-config");
+
+if (window.invenio) {
+  const uppyFilesApiClient = new UppyDepositFileApiClient({ apiHeaders }, defaultTransferType);
+  const uppyFilesService = new UppyDepositFilesService(rdmFilesApiClient, fileUploadConcurrency);
+
+  window.invenio.files = {
+    apiClient: uppyFilesApiClient,
+    service: uppyFilesService,
+    uploaderComponent: UppyUploader,
+  };
+}
+```
+
+For this to work, this configurable entrypoint needs to be included into deposit page *before* the
+entrypoint rendering the Deposit app.
+```jinja
+{# deposit.html #}
+{%- block javascript %}
+  {{ super() }}
++ {{ webpack[ config.FILES_UI_ENTRYPOINT ~ '.js'] }}
+  {{ webpack['invenio-app-rdm-deposit.js'] }}
+{%- endblock %}
+```
+
+#### Presentation/resources layer
+
+To let custom Files UI implementations know about available TransferTypes, we need to
+pass the information down to the `deposits-config` HTML input by extending the deposit's [get_form_config](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/views/deposits.py#L388) with:
+```python
+def get_form_config(**kwargs):
+    """Get the react form configuration."""
+   return dict(
+    #    ...
++       default_transfer_type=current_transfer_registry.default_transfer_type,
++       transfer_types=list(current_transfer_registry.get_transfer_types()),
+        **kwargs,
+    )
+```
+
+#### Deposit form app
+
+In `DepositFormApp` we drop the concept of always defaulting to `RDMDepositFiles*`.
+We default to using [services registered by Files UI entrypoint from above](https://github.com/oarepo/invenio-rdm-records/commit/a0adca214b279bff0f9ec546709b6fbb5aa5f39f) instead.
+
+Wherever we rendered the original `FileUploader` React component, we now
+[substitute it with the one from Files UI entrypoint](https://github.com/oarepo/invenio-app-rdm/commit/ca6959c2c3331825c73ced17a95086ccaf917e3e). There's still a possibility to override it, or its sub-component parts using
+`react-overriable`.
+
+#### Invenio-RDM-Records module exports
+
+To allow other Files UI implementation build on available Files Redux store actions, or possibly reusable
+file components (e.g. `FileListTable`), they now need to be exported by the `@js/invenio_rdm_records` module.
+
+### Custom Files UI with Uppy
+
+Here we describe integration of [Uppy.io](https://uppy.io/) as a custom Files UI implementation addressing
+use cases regarding reliable multi-part uploads.
+
+<!-- TODO: move under inveniosoftware org? -->
```

Comment:
Fine with us of course :)
I think there's some procedure to take care off for publishing on NPM.